src/PioneerUtils.jsm

```diff
@@ -164,7 +167,63 @@ export class PioneerUtils {
     const hashKey = `${this.config.studyName}/${await this.getPioneerId()}`;
     return sampling.chooseWeighted(this.config.branches, hashKey);
   }
+
+  /**
+   * Ends a study by uninstalling the addon and sending a relevant
+   * event ping to telemetry.
+   *
+   * @param {String?} eventId
+   *   The ID of the event that occured.
+   * 
+   * @returns {String}
+   *   The ID of the event ping that was submitted.
+   */
+  endStudy(eventId = PioneerUtils.EVENTS.ENDED_NEUTRAL) {
+    this.uninstall();
+    return this.submitEventPing(eventId);
+  }
+
+  /**
+   * Uninstalls the study addon.
+   *
+   * @returns {void}
+   */
+  async uninstall() {
+    const addon = await AddonManager.getAddonByID(this.config.addonId);
+    if (addon) {
+      addon.uninstall();
+    } else {
+      throw new Error(`Could not find addon with ID: ${this.config.addonId}`);
+    }
+  }
+
+  /**
+   * Submits an encrypted event ping.
+   *
+   * @param {String} eventId
+   *   The ID of the event that occured.
+   *
+   * @returns {String}
+   *   The ID of the event ping that was submitted.
+   */
+  submitEventPing(eventId) {
+    if (!(eventId in PioneerUtils.EVENTS)) {
+      throw new Error("Invalid event ID.");
```

Comment:
I don't think this will work. `foo in bar` when bar is an object checks if there is a key by the name of the value of foo in the object, so this would be checking if `"ended-neutral"` is a key in `PionerUtils.EVENTS`, which it isn't. I'm not sure how to do what you  want here. Maybe check `Object.values(PionerUtils.EVENTS)`?

Reply:
d'oh. yes! i had an array but i made it an object and forgot to accommodate that change.